include/service/entities/site_service.js

```diff
@@ -554,5 +569,54 @@ module.exports = function SiteServiceModule(pb) {
         return url.format(urlObject).replace(/\/$/, '');
     };
 
+    SiteService.prototype.deleteSiteSpecificContent = function (collections, siteid, callback) {
+        var self = this;
+
+        SiteService.siteExists(siteid, function (err, exists) {
+            if (util.isError(err)) {
+                return callback(err);
+            }
+
+            if (!exists) {
+                return callback(pb.BaseObjectService.validationError([pb.BaseObjectService.validationFailure("siteid", "Invalid siteid")]));
+            }
+
+            var tasks = util.getTasks(collections, function (collections, i) {
+                return function (taskCallback) {
+                    self.deleteSingle({where: {site: siteid}, collection: collections[i].name}, function (err, commandResult) {
```

Comment:
Good call. So we can just call dao.delete() here and siteService.deleteSingle below for the last event hook?